### ISSUE TITLE
Switching to Bootstrap 5 without JQuery

### ISSUE THREAD

**Issue by badstorm** [gobuffalo/cli#7]
With the introduction of Bootstrap 5 that drop support for JQuery, I'm trying to update a Buffalo project to only work with Vanilla JS.  Is there anyone trying to do the same?

I think we must rewrite _rails.js_ that depend on JQueryUI, unless there is already an updated version.  Another thing I'd like to do is introduce TypeScript support for programming the _assets/js/application.js_ file.

The question is: to do all this is it necessary only to update the _webpack.config.js_ and _package.js_ files? Does anyone already have some updated files?

If I get a good result I will send a PR as a proposal :)

**Comment by sio4** [comment]
Any update on this issue? I am interested in this.

For your question,

> where are located the template of the files generated when i run buffalo new?where are located the template of the files generated when i run buffalo new?

Yes, `genny/assets/webpack` is the place they are. (now `internal/genny/assets/webpack/templates/`) However, you need to update the `packr`. Please read `packrd/packed-packr.go` for your reference. It is a *zipped* storage of those templates. You can update it by executing command `packr2`.

Illustrate the workflow:

```console
$ go install github.com/gobuffalo/packr/v2/packr2@v2.8.1
$ cd .../cli
$ vi ...
$ packr2
$ git add internal/genny/assets/webpack/templates packrd/packed-packr.go
```

I hope this answer could help your work.

**Comment by badstorm** [comment]
Thanks @sio4 for the info. I'm trying to prepare a PR with my changes. Another question: how i can compile the cli? Before with the buffalo project I simply do `make install` but now if I do this command I get `make: *** No rule to make target `install'.  Stop.` and if i try `go build` nothing happend.  Thanks

**Comment by sio4** [comment]
For `cli` project, a common way of go works. From the root of the local working tree, please run `go install` command like following:

```console
$ pwd
/home/sio4/go/github.com/gobuffalo/cli
$ go install -v ./cmd/buffalo/
<...>
$ ls -l ~/go/bin/buffalo
-rwxrwxr-x 1 sio4 sio4 22394368  8월 30 21:27 /home/sio4/go/bin/buffalo
$ 
```

As the above output shows, the newly compiled binary will be installed and will be ready to run.

**Comment by badstorm** [comment]
Here my [PR](https://github.com/gobuffalo/cli/pull/23)

**Comment by sio4** [comment]
I submitted PR #27 and it was merged. By that PR, now the CLI generates Bootstrap 5 enabled skeleton with Webpack 5. I did not remove jQuery support though, since I believe jQuery is still the most popular javascript library and many popular modules still require it. However, users can easily remove the jQuery after the skeleton was generated.

I think supporting Typescript is also a good thing for buffalo.  I would like to suggest you convert this PR as a PR for Typescript support if you agree with the direction.
